src/Components/firebase-access.jsx

```diff
@@ -3,465 +3,386 @@ import "firebase/auth";
 import "firebase/firestore";
 import {db} from "../firebase";
 
-const STARTING_MONEY = 10000;
-
-
-export const addUser = (starting_money,gameId,nickname,numDays,numSymbols) => {
-  const gameRef = db.collection('Rooms').doc(gameId);
-  const userRef = gameRef.collection('users').doc();
-  const userId = userRef.id;
-  const empArray = Array.from(Array(numSymbols),()=>0);
-
-  const gameInfo = {
-      nickname: nickname,
-      net_worth: starting_money,
-      money_left: starting_money,
-      currentShares: empArray,
-  }
-  userRef.set(gameInfo);
-
-  for(var i_day = 0; i_day < numDays; i_day++) {
-      var newDoc = userRef.collection('investments').doc(i_day);
-      newDoc.set({
-          change: empArray,
-      });
-  }
-  return userId;
+const debug = (str) => {
 }
 
-// sets up the room to store symbols, indicator graphs, time series,
-// and default portfolio values
-export const setUpRoom = (db,symbolsL,Rounds,password) => {
-  const roomRef = db.collection('Rooms').doc();
-  roomRef.set({
-      day_index: 0,
-      phase: 'no-host',
-      password: password,
-  });
-  const roomID = roomRef.id;
-  initDates(db,symbolsL,Rounds).then((datesD)=> {
-      roomRef.update({
-          symbols: symbolsL,
-          dates: datesD["dates"],
-      });
-      initializeQuiz(symbolsL,roomID,datesD["period"],datesD["dates"]);
-  });
-  return roomID;
-}
+export const addUser = async (roomID,nickname) => {
+    const numDays = (await getDates(roomID)).length;
+    const numSymbols = (await getSymbols(roomID)).length;
+    const roomRef = await getRoomRef(roomID);
+    const userRef = await roomRef.collection('users').doc();
+    const userID = userRef.id;
+    const empArray = Array.from(Array(numSymbols),()=>0);
+    const starting_money = await getStartingMoney(roomID);
```

Comment:
use consistent naming conventions
```suggestion
    const startingMoney = await getStartingMoney(roomID);
```